diff --git a/kernel/kernel_umount.h b/kernel/kernel_umount.h
--- a/kernel/kernel_umount.h
+++ b/kernel/kernel_umount.h
@@ -8,8 +8,6 @@
 void ksu_kernel_umount_init(void);
 void ksu_kernel_umount_exit(void);
 
-#if !defined(CONFIG_KSU_SUSFS) || !defined(CONFIG_KSU_SUSFS_TRY_UMOUNT)
  // Handler function to be called from setresuid hook
  int ksu_handle_umount(uid_t old_uid, uid_t new_uid);
-#endif
 
 // for the umount list
 struct mount_entry {
diff --git a/kernel/kernel_umount.c b/kernel/kernel_umount.c
--- a/kernel/kernel_umount.c
+++ b/kernel/kernel_umount.c
@@ -109,6 +109,5 @@
 struct umount_tw {
 	struct callback_head cb;
 };
 
-#if !defined(CONFIG_KSU_SUSFS) || !defined(CONFIG_KSU_SUSFS_TRY_UMOUNT)
 static void umount_tw_func(struct callback_head *cb)
@@ -189,5 +189,4 @@
 	return 0;
 }
-#endif // #if defined(CONFIG_KSU_SUSFS) || !defined(CONFIG_KSU_SUSFS_TRY_UMOUNT)
 
 void ksu_kernel_umount_init(void)
