diff --git a/kernel/setuid_hook.c b/kernel/setuid_hook.c
--- a/kernel/setuid_hook.c
+++ b/kernel/setuid_hook.c
@@ -54,13 +54,11 @@ static inline bool is_zygote_normal_app_uid(uid_t uid)
 
 extern u32 susfs_zygote_sid;
 #ifdef CONFIG_KSU_SUSFS_SUS_PATH
-extern void susfs_run_sus_path_loop(uid_t uid);
+extern void susfs_run_sus_path_loop(void);
 #endif // #ifdef CONFIG_KSU_SUSFS_SUS_PATH
 #ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
-extern void susfs_reorder_mnt_id(void);
 #endif // #ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
 #ifdef CONFIG_KSU_SUSFS_TRY_UMOUNT
-extern void susfs_try_umount(uid_t uid);
 #endif // #ifdef CONFIG_KSU_SUSFS_TRY_UMOUNT
 #endif // #ifdef CONFIG_KSU_SUSFS
 
@@ -146,15 +146,14 @@ do_umount:
     // Handle kernel umount
 #ifndef CONFIG_KSU_SUSFS_TRY_UMOUNT
     ksu_handle_umount(old_uid, new_uid);
 #else
-    susfs_try_umount(new_uid);
+    ksu_handle_umount(old_uid, new_uid);
 #endif // #ifndef CONFIG_KSU_SUSFS_TRY_UMOUNT
 #ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
     // We can reorder the mnt_id now after all sus mounts are umounted
-    susfs_reorder_mnt_id();
 #endif // #ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
 
 #ifdef CONFIG_KSU_SUSFS_SUS_PATH
-    susfs_run_sus_path_loop(new_uid);
+    susfs_run_sus_path_loop();
 #endif // #ifdef CONFIG_KSU_SUSFS_SUS_PATH
 
diff --git a/kernel/kernel_umount.h b/kernel/kernel_umount.h
--- a/kernel/kernel_umount.h
+++ b/kernel/kernel_umount.h
@@ -8,8 +8,6 @@
 void ksu_kernel_umount_init(void);
 void ksu_kernel_umount_exit(void);
 
-#if !defined(CONFIG_KSU_SUSFS) || !defined(CONFIG_KSU_SUSFS_TRY_UMOUNT)
  // Handler function to be called from setresuid hook
  int ksu_handle_umount(uid_t old_uid, uid_t new_uid);
-#endif
 
 // for the umount list
 struct mount_entry {
diff --git a/kernel/kernel_umount.c b/kernel/kernel_umount.c
--- a/kernel/kernel_umount.c
+++ b/kernel/kernel_umount.c
@@ -109,7 +109,6 @@
 struct umount_tw {
 	struct callback_head cb;
 };
 
-#if !defined(CONFIG_KSU_SUSFS) || !defined(CONFIG_KSU_SUSFS_TRY_UMOUNT)
 static void umount_tw_func(struct callback_head *cb)
 {
@@ -189,5 +189,4 @@
 	return 0;
 }
-#endif // #if defined(CONFIG_KSU_SUSFS) || !defined(CONFIG_KSU_SUSFS_TRY_UMOUNT)
 
 void ksu_kernel_umount_init(void)
